lib/models/asset_token.dart

```diff
@@ -5,178 +5,370 @@
-    required this.source,
-    required this.sourceURL,
-    required this.thumbnailID,
-    required this.thumbnailURL,
-    required this.galleryThumbnailURL,
-    required this.title,
-    required this.initialSaleModel,
-    required this.ownerAddress,
-    required this.owners,
     required this.balance,
+    required this.owner,
     required this.lastActivityTime,
-    this.provenances,
-    this.updateTime,
-    this.isFeralfileFrame,
-    this.scrollable,
-    this.pending = false,
-    required this.originTokenInfoId,
-    this.swapped = false,
+    required this.lastRefreshedTime,
+    this.mimeType,
+    this.previewURL,
+    this.thumbnailURL,
+    this.thumbnailID,
+    this.galleryThumbnailURL,
+    this.pending,
+    this.isDebugged,
+    this.artistID,
+    this.blockchain,
+    this.tokenId,
+    this.title,
   });
 
-  factory AssetToken.fromAsset(Asset asset) => AssetToken(
-        artistName: asset.projectMetadata.latest.artistName,
-        artistURL: asset.projectMetadata.latest.artistUrl,
-        artistID: asset.projectMetadata.latest.artistId,
-        assetData: asset.projectMetadata.latest.assetData,
-        assetID: asset.projectMetadata.latest.assetId,
-        assetURL: asset.projectMetadata.latest.assetUrl,
-        basePrice: asset.projectMetadata.latest.basePrice,
-        baseCurrency: asset.projectMetadata.latest.baseCurrency,
-        blockchain: asset.blockchain,
-        blockchainUrl: asset.blockchainURL,
-        fungible: asset.fungible,
-        contractType: asset.contractType,
-        tokenId: asset.tokenId,
-        contractAddress: asset.contractAddress,
-        desc: asset.projectMetadata.latest.description,
-        edition: asset.edition,
-        editionName: asset.editionName,
-        id: asset.id,
-        maxEdition: asset.projectMetadata.latest.maxEdition,
-        medium: asset.projectMetadata.latest.medium,
-        mimeType: asset.projectMetadata.latest.mimeType,
-        mintedAt: asset.mintedAt.toIso8601String(),
-        previewURL: asset.projectMetadata.latest.previewUrl,
-        source: asset.projectMetadata.latest.source,
-        sourceURL: asset.projectMetadata.latest.sourceUrl,
-        thumbnailID: asset.thumbnailID,
-        thumbnailURL: asset.projectMetadata.latest.thumbnailUrl,
-        galleryThumbnailURL: asset.projectMetadata.latest.galleryThumbnailUrl,
-        title: asset.projectMetadata.latest.title,
-        initialSaleModel: asset.saleModel,
-        ownerAddress: asset.owner,
-        owners: asset.owners,
-        balance: asset.balance,
-        lastActivityTime: asset.lastActivityTime,
-        provenances: asset.provenance,
-        isFeralfileFrame:
-            asset.projectMetadata.latest.artworkMetadata?["isFeralfileFrame"],
-        pending: false,
-        originTokenInfoId: asset.originTokenInfo?.firstOrNull?.id,
-        swapped: asset.swapped ?? false,
-        scrollable: asset.attributes?.scrollable,
-      );
+  String id;
```

Comment:
I would like to make the fields use for compare `final` to completely prevent changing the order outside of the sorted list (prevent using `sortedList[0].id = newID` or `sortedList[0].lastActivityTime = newTime`).